crates/resolver/src/dns_lru.rs

```diff
@@ -42,6 +42,32 @@ impl LruValue {
     fn ttl(&self, now: Instant) -> Duration {
         self.valid_until.saturating_duration_since(now)
     }
+
+    fn with_updated_ttl(&self, now: Instant) -> Self {
+        let lookup = match self.lookup {
+            Ok(ref lookup) => {
+                let records = lookup
+                    .records()
+                    .iter()
+                    .map(|record| {
+                        let mut record = record.clone();
+                        record.set_ttl(self.ttl(now).as_secs() as u32);
```

Comment:
Looking at this, we probably want a Duration passed here instead of the u32 (can still store it as u32) that’s outside this change, but worth us looking into changing in the future.

Reply:
I opened #1995 for this